packages/react-components/react-carousel/library/src/components/Carousel/useCarousel.ts

```diff
@@ -81,16 +86,17 @@ export function useCarousel_unstable(props: CarouselProps, ref: React.Ref<HTMLDi
 
   const { announce } = useAnnounce();
 
-  const updateAnnouncement = useEventCallback(() => {
+  const updateAnnouncement = useEventCallback((newIndex: number) => {
```

Comment:
I updated this to take an index directly, since `activeIndex` isn't updated to the new value when `updateAnnouncement` is called